Added a FOSS License

AlexanderEvans opened the pull request:
I added a Free Open Source Software License to the project.  It is essentially the license with an extra paragraph for clarity about the applicability of the license.

Commit:
Added a FOSS Liscense

I added a Free Open Source Software Liscense to the project.  It is essentially the unliscense with an extra paragraph for clarity about the applicability of the liscense.

Dreaming381 reviewed (commented):
The FOSS license is great except that it completely removes your ability to use any Asset Store assets. Fortunately with the way Package Manager works you can still use Unity packages as is but you MUST have your Library folder in the gitignore and you can not embed packages so you can’t change the source code of packages.

AlexanderEvans committed:
Fixed license spelling

Fixed license spelling in the FOSS license.

AlexanderEvans commented on Unlicense Variation.txt:
```diff
@@ -0,0 +1,35 @@
+Please note that this license applies only to code and assets developed by 
+GCO for this project, not code or assets provided by Unity Technologies.  
+To obtain a liciense to the Unity Game Engine(If anyone wishes to use it to 
+compile or modify the game) they must obtain a lisence from Unity Technologies. 
```
Fixed

AlexanderEvans committed:
Changed the apostrophe charachter

Changed the apostrophe character, hopefully this one uses a more standard ASCII value.

AlexanderEvans commented on Unlicense Variation.txt:
```diff
@@ -0,0 +1,35 @@
+To obtain a liciense to the Unity Game Engine(If anyone wishes to use it to 
+compile or modify the game) they must obtain a lisence from Unity Technologies. 
+GCO does not have the licencing rights to redistribute the Unity Game 
+Engine or it�s source code, nor the ability to transfer our license to use 
```
Fixed.

AlexanderEvans committed:
Fixed Unity capitalization

Fixed Unity capitalization on line 34

AlexanderEvans commented on Unlicense Variation.txt:
```diff
@@ -0,0 +1,35 @@
+GCO does not have the licencing rights to redistribute the Unity Game 
+Engine or it�s source code, nor the ability to transfer our license to use 
+the Unity Game Engine, and thus grants no such rights.  Please see https://unity3d.com/ 
+for more information about getting your own copy of unity and licences 
```
fixed.

AlexanderEvans commented:
@Dreaming381 Hmmm... that is genuinely a big issue.  We could specifically state that we also explicitly differentiate the library folder from our code, and that it will need to be removed if it is pulled by someone?  I don't see us using paid assets, but I doubt the license is transferable so others that fork the repo should download it from unity... but I would really rather not have to remove it from the repo for internal purposes...  Hmmm....  Why is licensing so hard. 😝 I'll think about it though and if there is a simple way to resolve it.  Do you think it would be enough to add that clause to the license, or do you think that is pretty flimsy because it is still in the repo?  Or do you have another suggestion?  If we need to we *could* remove the library folder from the repo, but then keeping packages in sync across the team will become an even bigger headache...

AlexanderEvans commented:
I've assigned everyone involved in this game, active veteran members on git, and Dreaming381 who raised the issue to help discuss and resolve this licensing conundrum.  If we get this resolved here I will bring it up as an option for GCO going forward at a later meeting.  We have ignored the copyright and licensing of our projects for too long and I don't want this to suddenly come back and bite us in the butt one day.  Hopefully we can get some input from the group on this.

Dreaming381 commented:
> @Dreaming381 Hmmm... that is genuinely a big issue. We could specifically state that we also explicitly differentiate the library folder from our code, and that it will need to be removed if it is pulled by someone? I don't see us using paid assets, but I doubt the license is transferable so others that fork the repo should download it from unity... but I would really rather not have to remove it from the repo for internal purposes... Hmmm.... Why is licensing so hard. 😝 I'll think about it though and if there is a simple way to resolve it. Do you think it would be enough to add that clause to the license, or do you think that is pretty flimsy because it is still in the repo? Or do you have another suggestion? If we need to we _could_ remove the library folder from the repo, but then keeping packages in sync across the team will become an even bigger headache...

The library folder should be excluded by the .gitignore. Then the only way it gets pulled in is if someone messes with the .gitignore. The library is just a cache anyways. Anyone who forks the project will receive the package manager manifest.json and Unity will resolve the packages and download the packages for them. And it has versioning. The only downside is that you wouldn't be able to modify the package source code.

However, the bigger issue is that under FOSS, you cannot use FREE assets from the Asset Store. They are not public domain. The Asset Store Eula prohibits public distribution of assets unless it is the original creator releasing them under a different license. However, many assets provide an alternative license that are a little more lenient. And sometimes you can just ask the asset creator if it is ok.

AlexanderEvans commented:
Yeah, that means we would even lose things like textmesh pro, cinemachine, and the post-processing stack if the EULA is restrictive.  Is there some way to just exclude asset store files from the repo without having major issues with syncing the tools internally?

Dreaming381 commented:
Package Manager packages don't go through the asset store anymore. (The packages are also there but you should not download them that way.) Assuming you don't need to make custom modifications to the packages, as long as you gitignore the library folder, the package manifest.json will keep everything synced up for you. So that gives you access to TMP, Cinemachine, and PPS as well as all the other goodies that show up in the Package Manager window in Unity. 

Regarding the Asset Store, you'll have to forget it exists. There is a way to get around it, but I'm not even going to tell you because it is a 99.99% chance things will go very, very wrong if attempted.

AlexanderEvans commented:
I think that's fine then, as long as we can get those unity tools, especially PPS, I don't think we will really need any unity assets.  The only possible exception I can think of would be HD/realistic textures, and that is as simple as making a paidTextures folder and adding it to the .gitignore and keeping a copy of that in box, which while definitely not ideal, would probably work.  And that's only if we need to.  If possible I would rather have as many tools and assets made in house as possible anyway, so if we can avoid the asset store all together that's probably the best solution.
